cpp/oneapi/dal/algo/pca/backend/gpu/finalize_train_kernel_cov_dpc.cpp

```diff
@@ -36,6 +36,95 @@ using input_t = partial_train_result<task_t>;
 using result_t = train_result<task_t>;
 using descriptor_t = detail::descriptor_base<task::dim_reduction>;
 
+template <typename Float>
+auto compute_sums(sycl::queue& q,
+                  const pr::ndview<Float, 2>& data,
+                  const bk::event_vector& deps = {}) {
+    ONEDAL_PROFILER_TASK(compute_sums, q);
+    ONEDAL_ASSERT(data.has_data());
+    ONEDAL_ASSERT(data.get_dimension(1) > 0);
```

Comment:
I would recommend to follow the natural (left-to-right) order like `a < b && b < c && c <= d`

```suggestion
    ONEDAL_ASSERT(0 < data.get_dimension(1));
```